tests/ttnn/unit_tests/operations/test_moreh_softmin.py

```diff
@@ -397,3 +389,83 @@ def test_softmin_backward_optional_output_tensor(shape_dim, optional_output_tens
     passing, out = comp_allclose_and_pcc(x.grad, tt_dev, rtol=rtol, atol=atol)
     logger.info(out)
     assert passing
+
+
+@pytest.mark.parametrize(
+    "shape_dim",
+    [
+        [[32, 32], 1],  # H small
+        [[32, 32], 0],  # W small
+        [[2, 3, 32 * 4, 32 * 5], 3],  # H large
+        [[2, 3, 32 * 4, 32 * 5], 2],  # W large,
+        [[1, 15, 32, 32], 1],  # C
+    ],
+)
+def test_softmin_callback(shape_dim, device, use_program_cache):
+    shape, dim = shape_dim
+    torch.manual_seed(0)
+
+    x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16)
+
+    dev_x = ttnn.Tensor(x, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
+
+    tt_cpu = F.softmin(x, dim)
+    for i in range(2):
+        tt_npu = ttnn.operations.moreh.softmin(dev_x, dim)
+        if i == 0:
```

Comment:
You are still using the same `dev_x` in both loops. This way, the callback will produce the same output without any changes.
This way, the test will pass without implementing the callback.

Reply:
```Python
    for i in range(2):
        dev_x = ttnn.Tensor(x, ttnn.bfloat16).pad_to_tile(float("nan")).to(ttnn.TILE_LAYOUT).to(device)
        tt_npu = ttnn.operations.moreh.logsoftmax(dev_x, dim)
```

I have confirmed that you've modified `dev_x` to be created a new within the loop.
However, since the value of `dev_x` remains exactly the same as before, the test will pass even if the input address is not updated in the callback. This seems problematic.